components/sync/driver/brave_sync_service_impl.cc

```diff
@@ -175,4 +182,113 @@ SyncServiceCrypto* BraveSyncServiceImpl::GetCryptoForTests() {
+}  // namespace
+
+void BraveSyncServiceImpl::OnAccountDeleted(
+    const int current_attempt,
+    base::OnceCallback<void(const SyncProtocolError&)> callback,
+    const SyncProtocolError& sync_protocol_error) {
+  if (sync_protocol_error.error_type == SYNC_SUCCESS) {
+    std::move(callback).Run(sync_protocol_error);
+    // If request succeded - reset and clear all in a forced way
+    // The code below cleans all on an initiator device. Other devices in the
+    // chain will be cleaned at BraveSyncServiceImpl::ResetEngine
+    DCHECK(initiated_delete_account_);
+    BraveSyncServiceImpl::StopAndClear();
+  } else if (current_attempt < kMaxPermanentlyDeleteSyncAccountAttempts) {
+    // Server responded failure, but we need to try more
+    base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
+        FROM_HERE,
+        base::BindOnce(&BraveSyncServiceImpl::PermanentlyDeleteAccountImpl,
+                       weak_ptr_factory_.GetWeakPtr(), current_attempt + 1,
+                       std::move(callback)),
+        base::Milliseconds(kDelayBetweenDeleteSyncAccountAttemptsMsec));
+  } else {
+    // Server responded failure, and we are out of our attempts
+    initiated_delete_account_ = false;
+    std::move(callback).Run(sync_protocol_error);
+  }
+}
+
+void BraveSyncServiceImpl::PermanentlyDeleteAccountImpl(
+    const int current_attempt,
+    base::OnceCallback<void(const SyncProtocolError&)> callback) {
+  if (!engine_) {
+    // We can reach here if two devices almost at the same time will initiate
+    // the deletion procedure
+    SyncProtocolError sync_protocol_error;
+    sync_protocol_error.error_type = SYNC_SUCCESS;
+    std::move(callback).Run(sync_protocol_error);
+    return;
+  }
+
+  DCHECK_GE(current_attempt, 1);
+  DCHECK_NE(current_attempt, 10);
+
+  engine_->PermanentlyDeleteAccount(base::BindOnce(
+      &BraveSyncServiceImpl::OnAccountDeleted, weak_ptr_factory_.GetWeakPtr(),
+      current_attempt, std::move(callback)));
+}
+
+void BraveSyncServiceImpl::PermanentlyDeleteAccount(
+    base::OnceCallback<void(const SyncProtocolError&)> callback) {
+  initiated_delete_account_ = true;
+  PermanentlyDeleteAccountImpl(1, std::move(callback));
+}
+
+void BraveSyncServiceImpl::ResetEngine(ShutdownReason shutdown_reason,
+                                       ResetEngineReason reset_reason) {
+  SyncServiceImpl::ResetEngine(shutdown_reason, reset_reason);
+
+  if (initiated_self_device_info_deleted_) {
+    return;
+  }
+
+  if (shutdown_reason == ShutdownReason::DISABLE_SYNC_AND_CLEAR_DATA &&
+      reset_reason == ResetEngineReason::kDisabledAccount &&
+      sync_disabled_by_admin_ && !initiated_delete_account_ &&
+      !initiated_join_chain_) {
+    brave_sync_prefs_.SetSyncAccountDeletedNoticePending(true);
+    // Forcing stop and clear, because sync account was deleted
+    BraveSyncServiceImpl::StopAndClear();
+    return;
+  }
+
+  if (shutdown_reason == ShutdownReason::DISABLE_SYNC_AND_CLEAR_DATA &&
```

Comment:
`else if` and remove the return

Reply:
fixed at 
https://github.com/brave/brave-core/pull/13562/commits/01ae0cb6d5a61d27be9abfffafe56c3587ff3fba